packages/firestore/src/api/database.ts

```diff
@@ -2487,7 +2274,7 @@ export class CollectionReference<T = DocumentData>
   }
 }
 
-function validateSetOptions(
+export function validateSetOptions(
```

Comment:
Doesn't this potentially create a cycle? Better to move this into `input_validation.ts`.

Reply:
Done